mkdocs/docs/meili/thoughts_schema_interpreter_v,0,0,1.md

```diff
@@ -0,0 +1,288 @@
+# cloece — TS → JSON Manifest (MVP)
+
+## Goal
+
+Turn a **TypeScript** input into a single **JSON manifest** describing:
+
+* **Entities** → D1 tables + fields
+* **CRUD Routes** → Workers
+* **Resource list** → D1 + R2 buckets
+
+---
+
+## Scope
+
+* Single database (for now). User defines the DB name.
+* **Decorators supported:** `@D1`, `@R2`, `@Workers.GET`, `@Workers.POST`
+* **Types supported:** float, int, string, boolean, Date, R2Id
+* **Out of scope (v0):** relations, indexes, uniques, defaults, auth, hashing, watch mode, migrations.
+
+---
+
+## Decorators (MVP)
+
+```ts
+@D1(options?)              // class: marks an entity
+// options: { table?: string }  // optional table override
+
+@R2({ bucket: string })    // field: marks an R2 object key
+
+@Workers.GET(path?)        // method: emits a GET route
+@Workers.POST(path?)       // method: emits a POST route
+// If path omitted → "/<entity>/<methodName>"
+```
+
+---
+
+## Type Mapping
+
+| TS type | IR `type`  | Notes                      |
+| ------: | ---------- | -------------------------- |
+|  number | int        |                            |
+|  string | text       |                            |
+| boolean | bool       |                            |
+|    Date | date       | store as ISO string        |
+|    R2Id | r2\_object | requires `@R2({ bucket })` |
+
+**Nullability:** a field is nullable if it has `?` or is unioned with `null`/`undefined`.
+**Primary key rule:** if a numeric field named `id` exists → `pk: true`, `auto: true`. Otherwise: **error** (v0 only supports `id: number` as PK).
+
+---
+
+## JSON Manifest
+
+> **Note**
+> Subject to change. Since all of cloece’s functionality depends on the manifest, we may tweak it to fit the Rust compiler’s needs. The POST request syntax is a bit verbose, but we assume users won’t hand-edit this JSON.
+
+```json
+{
+  "version": "0.0.1",
+  "project": "my-app",
+  "entities": [
+    {
+      "name": "Person",
+      "table": "person",
+      "fields": [
+        { "name": "id", "type": "int", "pk": true, "auto": true, "nullable": false },
+        { "name": "name", "type": "text", "nullable": false },
+        { "name": "age", "type": "int", "nullable": false },
+        { "name": "image", "type": "r2_object", "bucket": "images", "nullable": true }
```

Comment:
for `"type":` I would be fine using an agreed upon enum rather than switching based off an string, like
`"type": 0` and so on